### PR TITLE
test: add node updater pool unit tests

### LINKED ISSUE #1251: Implement unit tests for the Node Updater Pool
**What would you like to be added**:
When the Node Updater Pool was introduced in the code, we didn't have unit tests covering all parts of the new structure. It would be great to add these unit tests, to improve the code coverage.

**Why is this needed**:
Improve code coverage and users' confidence in the nfd project.

### PR DESCRIPTION
**Pull request by AhmedGrati**
Resolves #1251.

### CODE REVIEW

**Review by ArangoGutierrez — APPROVED** [review]
```bash
[eduardo@fedora node-feature-discovery]$ make test
go test -covermode=atomic -coverprofile=coverage.out ./cmd/... ./pkg/... ./source/...
?       sigs.k8s.io/node-feature-discovery/cmd/nfd-master       [no test files]
ok      sigs.k8s.io/node-feature-discovery/cmd/nfd-topology-gc  0.009s  coverage: 33.3% of statements
?       sigs.k8s.io/node-feature-discovery/pkg/apihelper        [no test files]
?       sigs.k8s.io/node-feature-discovery/pkg/cpuid    [no test files]
?       sigs.k8s.io/node-feature-discovery/pkg/generated/clientset/versioned    [no test files]
?       sigs.k8s.io/node-feature-discovery/pkg/generated/clientset/versioned/fake       [no test files]
?       sigs.k8s.io/node-feature-discovery/pkg/generated/clientset/versioned/scheme     [no test files]
?       sigs.k8s.io/node-feature-discovery/pkg/generated/clientset/versioned/typed/nfd/v1alpha1 [no test files]
?       sigs.k8s.io/node-feature-discovery/pkg/generated/clientset/versioned/typed/nfd/v1alpha1/fake    [no test files]
?       sigs.k8s.io/node-feature-discovery/pkg/generated/informers/externalversions     [no test files]
?       sigs.k8s.io/node-feature-discovery/pkg/generated/informers/externalversions/internalinterfaces  [no test files]
?       sigs.k8s.io/node-feature-discovery/pkg/generated/informers/externalversions/nfd [no test files]
?       sigs.k8s.io/node-feature-discovery/pkg/generated/informers/externalversions/nfd/v1alpha1        [no test files]
?       sigs.k8s.io/node-feature-discovery/pkg/generated/listers/nfd/v1alpha1   [no test files]
?       sigs.k8s.io/node-feature-discovery/pkg/labeler  [no test files]
ok      sigs.k8s.io/node-feature-discovery/cmd/nfd-topology-updater     0.031s  coverage: 30.8% of statements
ok      sigs.k8s.io/node-feature-discovery/cmd/nfd-worker       0.013s  coverage: 66.1% of statements
ok      sigs.k8s.io/node-feature-discovery/pkg/apis/nfd/v1alpha1        0.008s  coverage: 21.8% of statements
?       sigs.k8s.io/node-feature-discovery/pkg/nfd-topology-updater/kubeletnotifier     [no test files]
?       sigs.k8s.io/node-feature-discovery/pkg/podres   [no test files]
?       sigs.k8s.io/node-feature-discovery/pkg/podres/mocks     [no test files]
?       sigs.k8s.io/node-feature-discovery/pkg/utils/hostpath   [no test files]
?       sigs.k8s.io/node-feature-discovery/pkg/version  [no test files]
?       sigs.k8s.io/node-feature-discovery/source/custom        [no test files]
?       sigs.k8s.io/node-feature-discovery/source/custom/rules  [no test files]
?       sigs.k8s.io/node-feature-discovery/source/fake  [no test files]
ok      sigs.k8s.io/node-feature-discovery/pkg/nfd-master       13.065s coverage: 52.6% of statements
ok      sigs.k8s.io/node-feature-discovery/pkg/nfd-topology-gc  3.419s  coverage: 48.5% of statements
ok      sigs.k8s.io/node-feature-discovery/pkg/nfd-topology-updater     0.013s  coverage: 6.2% of statements
ok      sigs.k8s.io/node-feature-discovery/pkg/nfd-worker       3.111s  coverage: 68.4% of statements
ok      sigs.k8s.io/node-feature-discovery/pkg/resourcemonitor  0.048s  coverage: 75.7% of statements
ok      sigs.k8s.io/node-feature-discovery/pkg/topologypolicy   0.007s  coverage: 100.0% of statements
ok      sigs.k8s.io/node-feature-discovery/pkg/utils    0.006s  coverage: 18.9% of statements
ok      sigs.k8s.io/node-feature-discovery/pkg/utils/kubeconf   0.025s  coverage: 30.6% of statements
ok      sigs.k8s.io/node-feature-discovery/source       0.006s  coverage: 22.4% of statements
ok      sigs.k8s.io/node-feature-discovery/source/cpu   0.006s  coverage: 6.9% of statements
ok      sigs.k8s.io/node-feature-discovery/source/kernel        0.005s  coverage: 9.1% of statements
ok      sigs.k8s.io/node-feature-discovery/source/local 0.006s  coverage: 8.4% of statements
ok      sigs.k8s.io/node-feature-discovery/source/memory        0.004s  coverage: 21.6% of statements
ok      sigs.k8s.io/node-feature-discovery/source/network       0.004s  coverage: 16.7% of statements
ok      sigs.k8s.io/node-feature-discovery/source/pci   0.007s  coverage: 24.4% of statements
ok      sigs.k8s.io/node-feature-discovery/source/storage       0.006s  coverage: 25.0% of statements
ok      sigs.k8s.io/node-feature-discovery/source/system        0.006s  coverage: 22.2% of statements
ok      sigs.k8s.io/node-feature-discovery/source/usb   0.003s  coverage: 23.3% of statements
```

leaving @marquiz for final approval

**Comment by AhmedGrati** [comment]
ping @marquiz

File: pkg/nfd-master/node-updater-pool_test.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+)
+
+func newMockNodeUpdaterPool(nfdMaster *nfdMaster) *nodeUpdaterPool {
+	return &nodeUpdaterPool{
+		nfdMaster: nfdMaster,
+		wg:        sync.WaitGroup{},
+	}
+}
+
+func TestStart(t *testing.T) {
```
Comment (marquiz) [inline pkg/nfd-master/node-updater-pool_test.go:38]: nit about the naming, be a bit more specific about which type's start method are we testing

```suggestion
func TestNodeUpdaterStart(t *testing.T) {
```

File: pkg/nfd-master/node-updater-pool_test.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+		})
+
+		mockNodeUpdaterPool.start(10)
+		Convey("Node updater pool queue should not change", func() {
+			So(mockNodeUpdaterPool.queue, ShouldEqual, q)
+		})
+	})
+}
+
+func TestStop(t *testing.T) {
```
Comment (marquiz) [inline pkg/nfd-master/node-updater-pool_test.go:58]:
```suggestion
func TestNodeUpdaterStop(t *testing.T) {
```

**Review by marquiz — APPROVED** [review]
Thanks @AhmedGrati for improving the test coverage
/assign @fmuyassarov

**Comment by ArangoGutierrez** [comment]
PING @fmuyassarov

**Comment by AhmedGrati** [comment]
ping @fmuyassarov

**Comment by ArangoGutierrez** [comment]
ping @fmuyassarov

**Pull request merged**